Automattic/wp-calypso
# Data Layer: Update timezones to use wpcom-http

### Changes introduced by this PR

* Update timezones data-layer handler to use wpcom-http instead of `wpcom.js`.
* Removes `isRequesting` flag usage from `<QueryTimezones />` component.
* Removes `isRequestingTimezones` selector
* Removes success/failure actions for timezones requesting actions and reducer.

Fixes #17826.

#### Testing instructions

1. Check this branch.
1. Open the console.
1. Run `dispatch( { type: 'TIMEZONES_REQUEST' } );`.
1. Switch to the Redux Dev Tools Panel.
1. Verify you can see the timezones network request.
1. Verify the `TIMEZONES_RECEIVE` action gets dispatched with the received timezones.
1. Run `npm run test-client client/state/data-layer/wpcom/timezones`
1. Expect to see the tests pass
